drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationUtils.java

```diff
@@ -75,6 +80,18 @@ public static boolean isSimpleJavaType(String className) {
                 .anyMatch(className::equals);
     }
 
+    /**
+     * It defines if a set <code>FactMapping</code> represents an <b>Expression type</b>.
+     * This is TRUE if: the Fact class is NOT simple and there is ONLY ONE step into
+     * ExpressionElements list (which is the Fact).
+     * @param factMapping
+     * @return
+     */
+    public static boolean isExpressionType(FactMapping factMapping) {
+        return !isSimpleJavaType(factMapping.getFactIdentifier().getClassName()) &&
+                factMapping.getExpressionElements().size() == 1;
```

Comment:
Is it really needed this additional condition? If we are able to remove it, it will be easier to support expressions on not root complex type

Reply:
@danielezonca I think I found an alternative: analyzing the scesim file, I discovered that we can check if FactIdentifer.className = FacrMapping.className (and className is not a Simple type) could be a reasonable alternative. If the class of the property is equal to  the class of fact and the class is complex type, then is an expression type.

Reply:
@danielezonca At the moment I didn't find a different way to infer that information from the current data.